### DISCUSSION TITLE
Is there a way I can check/reset caps lock status ?

### DISCUSSION THREAD

**Discussion by 1bonbrie** [hathach/tinyusb#2361] [General]
Hello,

I could write a code that turns on and off the caps lock LED when the caps lock key is pressed on a keyboard. But if I press the caps lock key on (turning the LED on),  disconnect and reconnect my MCU, the LED is off while the caps lock is active on the computer. Would it be possible to check its status/reset the status to inactive ?

**Comment by ObligedTester** [comment]
Hello,

Someone feel free to correct me if I'm wrong.
As I understand it, the keyboard do not care and do not keep track off current caps lock status. This is up to the MCU or the software to activate the LED again after a keyboard is connected for example.

Are you writing a host program?

**Reply by 1bonbrie** [reply]
Yes, I am writing a host program. Yes the keyboard of course does not track anything. A more appropriate way to ask my question would be is there a way to check the led bit right when the keyboard is connected. Because tinyusb will only check the led bit when hid_report_callback is called (which is not called right when you connect the keyboard). I do not know if there is a way I could hardcode a callback. Maybe make the code press shift key would be the easiest solution?

**Reply by hathach** [reply]
when a new keyboard enumerated, all of its indicator (numlock, caplock) should be default to off. If you are writing a host program, you must keep track of the current indicator, and set them if they are currently on.
